frigate/record/cleanup.py

```diff
@@ -215,34 +215,62 @@ def sync_recordings(self) -> None:
         logger.debug("Start sync recordings.")
 
         # get all recordings in the db
-        recordings: Recordings = Recordings.select()
-
-        # get all recordings files on disk
-        files_on_disk = []
-        for root, _, files in os.walk(RECORD_DIR):
-            for file in files:
-                files_on_disk.append(os.path.join(root, file))
-
-        recordings_to_delete = []
-        for recording in recordings.objects().iterator():
-            if recording.path not in files_on_disk:
-                recordings_to_delete.append(recording.id)
+        recordings = Recordings.select(Recordings.id, Recordings.path)
+
+        # get all recordings files on disk and put them in a set
+        files_on_disk = {
+            os.path.join(root, file)
+            for root, _, files in os.walk(RECORD_DIR)
+            for file in files
+        }
+
+        # Use pagination to process records in chunks
+        page_size = 1000
+        num_pages = (recordings.count() + page_size - 1) // page_size
+        recordings_to_delete = set()
+
+        for page in range(num_pages):
+            for recording in recordings.paginate(page, page_size):
+                if recording.path not in files_on_disk:
+                    recordings_to_delete.add(recording.id)
+
+        # convert back to list of dictionaries for insertion
+        recordings_to_delete = [
+            {"id": recording_id} for recording_id in recordings_to_delete
+        ]
+
+        if len(recordings_to_delete) / recordings.count() > 0.5:
+            logger.debug(
```

Comment:
Maybe we should make this log a warning to alert the user about potential misconfiguration for example in the event that their NAS mount failed?

Reply:
I believe that notifying user about the disconnected NAS should be placed elsewhere, definitely not in a internal utility function. 🙂